### ISSUE TITLE
False positive unsupported-assignment-operation with pandas

### ISSUE THREAD

**Issue by sam-s** [pylint-dev/pylint#9037]
### Bug description

`pylint` reports `unsupported-assignment-operation`  here:
```python
"https://github.com/pylint-dev/pylint/issues/9037"

import pandas as pd

df = pd.DataFrame({"a":[1,2,3],"b":[1,2,3]}).set_index("a")
df["asdf"] = [1,2,3]
```
note that if I remove `.set_index("a")`, the error is gone.

### Command used

```shell
pylint pylint-9037.py
```

### Pylint output

```shell
pylint-9037.py:6:0: E1137: 'df' does not support item assignment (unsupported-assignment-operation)
```

### Expected behavior

no diagnostics

### Pylint version

```shell
pylint 2.17.5
astroid 2.15.6
Python 3.11.5 (tags/v3.11.5:cce6ba9, Aug 24 2023, 14:38:34) [MSC v.1936 64 bit (AMD64)]
```

### OS / Environment

msys2

### Additional dependencies

pandas==2.0.3

**Comment by mbyrnepr2** [comment]
I think this looks like a duplicate of https://github.com/pylint-dev/pylint/issues/7564. Would you agree @sam-s? Thanks for raising it by the way :).
